### DISCUSSION TITLE
Can I use FSP for EigenValue calculation

### DISCUSSION THREAD

**Discussion by js-jixu** [idaholab/moose#22728] [Q&A General] [answered]
Hi, experts.

1. 
I'm using `InversePowerMethod` to do some eigenvalue calculation. When the problem size is large, the computational cost of using `SMP` will be high. So can `FSP` be used in the eigenvalue calculation of MOOSE (just like the steady or transient calculation)？

2. 
I found that compared to `InversePowerMethod`, MOOSE now recommends `Eigenvalue` for eigenvalue calculation. I now have an `Executioner` for `InversePowerMethod` that I did before. But when I wanted to transform it from `InversePowerMethod` to `Eigenvalue`, I found some differences between `InversePowerMethod` and `Eigenvalue`. For example, in `InversePowerMethod`, I will set `xdiff` and `bnorm`, but these are not available in `Eigenvalue`. How can this be solved?

**Comment by GiudGiud** [comment, marked as answer]
Hello

1) I dont know that it's been used for eigenvalue calculations

2) This will require additional development to provide these settings in our new eigenvalue executioner. 

Guillaume

**Reply by GiudGiud** [reply]
For 2 you should be able to reproduce the previous behavior using the 
- normalization postprocessor to indicate what computes the norm
- normal_factor : the norm to reach after normalization

**Reply by GiudGiud** [reply]
@YaqiWang will know more than me about normalization of the output of the eigenvalue executioner

**Reply by js-jixu** [reply]
I have used `Eigenvalue` and `NonlinearEigen`. `NonlinearEigen` can run but the convergence is not good, and `Eigenvalue` can't run.

```
# [Executioner]
#   type = NonlinearEigen
#   free_power_iterations = 6

#   normalization = 'powernorm'
#   normal_factor = 21230

#   bx_norm = 'bnorm'
#   k0 = 1.3
#   l_max_its = 200
#   nl_abs_tol = 1e-7

#   solve_type = 'NEWTON'
#   petsc_options = '-snes_converged_reason -ksp_converged_reason -snes_linesearch_monitor'
#   petsc_options_iname = '-pc_type -pc_factor_shift_type -pc_factor_mat_solver_package'
#   petsc_options_value = 'lu       NONZERO               superlu_dist'

#   line_search = none
#   automatic_scaling = true
# []

# [Executioner]
#   type = Eigenvalue
#   free_power_iterations = 6

#   normalization = 'powernorm'
#   normal_factor = 21230

#   initial_eigenvalue = 1
#   l_max_its = 200
#   nl_abs_tol = 1e-7

#   solve_type = 'NEWTON'
#   petsc_options = '-snes_converged_reason -ksp_converged_reason -snes_linesearch_monitor'
#   petsc_options_iname = '-pc_type -pc_factor_shift_type -pc_factor_mat_solver_package'
#   petsc_options_value = 'lu       NONZERO               superlu_dist'

#   line_search = none
#   automatic_scaling = true
# []
```

So I think I'd better use InversePowerMethod.

**Reply by js-jixu** [reply]
1. When I use `Eigenvalue`, I meet a `Segmentation fault (signal 11)`.

2. I want to know if [this change](https://github.com/idaholab/moose/pull/22643) has been merged into the latest moose, I want to wait until it is merged to update moose and do some FSP testing.
